lib/gollum-lib/filter/critic_markup.rb

```diff
@@ -0,0 +1,93 @@
+# ~*~ encoding: utf-8 ~*~
+
+# CriticMarkup
+#
+# Render CriticMarkup
+
+class Gollum::Filter::CriticMarkup < Gollum::Filter
+
+  # Patterns inspired by https://github.com/DivineDominion/criticmarkup.tmbundle/blob/master/Syntaxes/criticmarkup.tmLanguage
+  # All patterns use multiline matching (m flag)
+  # Logic inspired by https://github.com/CriticMarkup/CriticMarkup-toolkit/blob/master/CLI/criticParser_CLI.py
+  
+  ADDITION_PATTERN      = %r|{\+\+(?<content>.*?)\+\+[ \t]*(\[(.*?)\])?[ \t]*\}|m
+  DELETION_PATTERN      = %r|{--(?<content>.*?)--[ \t]*(\[(.*?)\])?[ \t]*\}|m
+  SUBSTITUTION_PATTERN  = %r|{~~(?<oldcontent>.*?)~>(?<newcontent>.*?)~~}|m
+  HIGHLIGHT_PATTERN     = %r|{\=\=(?<content>.*?)[ \t]*(\[(.*?)\])?[ \t]*\=\=\}{>>(?<comment>.*?)<<}|m
+  COMMENT_PATTERN       = %r|{>>(?<content>.*?)<<}|m
+
+  PROCESS_PATTERN       = /(?<placeholder>CRITIC\h+)/
```

Comment:
> What does \h match? Can't find in on rubular.com (which is my regexp cheat sheet).

`\h` matches a hexdigit character ([0-9a-fA-F]).

Reply:
> Below placeholder is set as "CRITIC" + bla, here the placeholder catchgroup is the bit before "CRITIC" rather than after it. How does that end up matching each other?

The `?<name>` syntax just names the match group. The actual matching syntax follows naming (within the parentheses). See the Emoji filter for another example of this.

Reply:
> maybe PROCESS_PATTERN should be more specific

Would it help if it had a specific length (e.g., 40 hex characters)?

Reply:
> The ?<name> syntax just names the match group. The actual matching syntax follows naming (within the parentheses). See the Emoji filter for another example of this.

Thanks, that explains it! So the magic is in the `\h`. Yes, you could just match exactly 40 hex characters then. Unless you think that would make it much slower. Alternatively, you can just make the pattern a bit more exotic, like ==CRITIC or something (the emoji filter has something like that).

Reply:
Fixed in https://github.com/gollum/gollum-lib/pull/308/commits/0d5c4d6699aab073b143c15de8ed695886373e0f.